File: 社区相关/LeetCode/1109.航班预订统计.cpp
/*
 * @lc app=leetcode.cn id=1109 lang=cpp
 *
 * [1109] 航班预订统计
 */

// @lc code=start
class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
        vector<int> nums(n);
        for (auto& booking : bookings) {
            int i = booking[0];
            int j = booking[1];
            int seats = booking[2];
            nums[i - 1] += seats;
            if (booking[1] < n) nums[j] -= seats;
        }
        for (int i = 1; i < n; i++) {
            nums[i] += nums[i - 1];
        }
        return nums;
    }
};


// @lc code=end

File: 社区相关/LeetCode/CodeStash/1109.航班预订统计.py
#
# @lc app=leetcode.cn id=1109 lang=python3
#
# [1109] 航班预订统计
#

# @lc code=start
from itertools import accumulate
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        ans = [0] * n
        for i, j, seats in bookings:
            ans[i-1] += seats
            if j < n:
                ans[j] -= seats 
        ans = list(accumulate(ans))
        return ans
# @lc code=end

